### DISCUSSION TITLE
Release 2.0.0b14 - beta quality, for testing only

### DISCUSSION THREAD

**Discussion by ThomasWaldmann** [borgbackup/borg#8548] [General]
borgbackup 2.0.0b14 beta release - do not use this on production backup repositories.

Long changelog:
https://borgbackup.readthedocs.io/en/2.0.0b14/changes.html#version-2-0-0b14-2024-11-17

If you use pip to install this, use: pip install "borgbackup==2.0.0b14"

Docs: https://borgbackup.readthedocs.io/en/2.0.0b14/

Discussion:

Notes:
- Binaries will come later.
- Only limited platform testing and binaries this time, see "vagrant" tag in issue tracker.

<hr /><em>This discussion was created from the release <a href='https://github.com/borgbackup/borg/releases/tag/2.0.0b14'>Release 2.0.0b14 - beta quality, for testing only</a>.</em>

**Comment by ThomasWaldmann** [comment]
Due to a small bug, the b14 usage docs do not link to the `borg undelete` part, it is there (it can be also found by searching for "undelete"):

https://borgbackup.readthedocs.io/en/2.0.0b14/usage/undelete.html

**Comment by bjo81** [comment]
While trying to transfer a repo from 1.x to 2, I ran into a "no space left on device" issue:

```
borgstore.backends.errors.BackendError: rclone rc command failed: error 500: {                                                                                                                
        "error": "write /root/pd/mail/data/0b/35/0b353630e41a18784ad2de02ebade55c55b8e63cc087838f51faea3b05e761bf: no space left on device",                                                  
        "input": {                                                                                                                                                                            
                "fs": "pd/mail/",                                                                                                                                                             
                "remote": "data/0b/35"                                                                                                                                                        
        },                                                                                                                                                                                    
        "path": "operations/uploadfile",                                                                                                                                                      
        "status": 500                                                                                                                                                                         
}                                                                                                                                                                                             
                         
```
This is imho somehow strange as the command was
`borg2 transfer --repo rclone:pd/mail  --from-borg1 --other-repo /tmp/hi/mail`

and pd is a "pixeldrain" backend, so no local storage. And rclone:pd schould be correct since b13.

**Reply by ThomasWaldmann** [reply]
So it stored everything into local `/root/pd` folder?

What's your borgstore version?

`pd` is a configured remote for rclone when you invoke it as root?

**Reply by bjo81** [reply]
> So it stored everything into local `/root/pd` folder?

Yes.

> What's your borgstore version?

I don't know, that one bundled in b14.

> `pd` is a configured remote for rclone when you invoke it as root?

Yes.

**Reply by ThomasWaldmann** [reply]
borgstore is not bundled into borg, but automatically installed via dependencies.
if you used pip to install, check `pip freeze` output.

**Reply by ThomasWaldmann** [reply]
Hmm, guess the rclone spec isn't right.

rclone would use `pd:mail`, so for borg it would be `rclone:pd:mail`.

when giving `pd/mail` to rclone, it thinks you want to use a local fs path.

**Reply by bjo81** [reply]
> borgstore is not bundled into borg, but automatically installed via dependencies. if you used pip to install, check `pip freeze` output.

I meant bundled within the binary.

**Reply by bjo81** [reply]
> Hmm, guess the rclone spec isn't right.
> 
> rclone would use `pd:mail`, so for borg it would be `rclone:pd:mail`.
> 
> when giving `pd/mail` to rclone, it thinks you want to use a local fs path.

This is IMHO confusing. In case of S3 storages, rclone:remote:bucket would be the bucket. In other cases it's just the subdirectory?

**Reply by ThomasWaldmann** [reply]
It's the way how rclone expects it, see the rclone docs.

**Reply by bjo81** [reply]
My bad.
Trying to use a sftp backend now: `sftp://u411047@u411047.your-storagebox.de:23/mail`, which throws
`Error: Invalid Backend Storage URL:`
I stumpled upon "Docs: user@ and :port are optional in sftp and ssh URLs." regarding b11, but it's not mentioned here: https://borgbackup.readthedocs.io/en/2.0.0b11/ - unfortunately it's also not mentioned here: https://borgbackup.readthedocs.io/en/2.0.0b14/quickstart.html#remote-repositories

**Reply by ThomasWaldmann** [reply]
That URL looks valid, these are the docs about repo URLs:

https://borgbackup.readthedocs.io/en/2.0.0b14/usage/general.html#repository-urls

If you can reproduce that "invalid backend storage url" with b14, file a bug report with the full commandline and full output.

The docs you linked just gives an example of a ssh:// remote repo, not sftp.

User and port being optional means that you can leave them away and it will use default values:
- the default port (22)
- the current local logged-in user name

**Reply by bjo81** [reply]
Seems like I searched in the wrong place.

**Comment by dietmargoldbeck** [comment]
### traceback with 2.0.0b14 at start of backup

At the start of each backup with 2.0.0b14 i have this traceback. After the traceback the backup runs normal
```
security: read manifest timestamp '2024-11-29T02:27:33.404882+00:00'
security: determined newest manifest timestamp as 2024-11-29T02:27:33.404882+00:00
security: repository checks ok, allowing access
Processing files ...
0 B O 0 B U 0 N .                                                               ^MReading files cache ...
Traceback (most recent call last):

  File "borg/remote.py", line 266, in inner_serve

  File "borg/repository.py", line 539, in store_delete

  File "borgstore/store.py", line 218, in delete

  File "borgstore/backends/posixfs.py", line 157, in delete

borgstore.backends.errors.ObjectNotFound: cache/chunks_hash

Traceback (most recent call last):

  File "borg/remote.py", line 266, in inner_serve

  File "borg/repository.py", line 539, in store_delete

  File "borgstore/store.py", line 218, in delete

  File "borgstore/backends/posixfs.py", line 157, in delete

borgstore.backends.errors.ObjectNotFound: cache/chunks

cached chunk indexes: ['0b818386e736f02b', '183453c5584142bc', '1f1a8afdc79eda74', '29713fb1f54f5fc3', '2f2a64a0799cfc41', '47b23c61ed9866d1', '613fc10119910e8d', '63dca754d51d4c08', '770613bdb7f07db5', '880a2f48e02c354a', '8b91de2ab3cb8ab6', '93a9f38eb3aa71b3', 'aae304a7dce88d3c', 'ad64b7b464aa5e20', 'b3f9bf7ae273c33f', 'b56538401a81aa2f', 'bffde67d669e229d', 'cde393ccc3cb8787', 'ce34f721671b32b2', 'd030d64cd6968857', 'de694a511e9ecd05', 'f057e41bc229c56e', 'f42ce82301d7ed46', 'f64337d288c928cd', 'f9e544f77f0bd246']
trying to load cache/chunks.0b818386e736f02b from the repo...
cache/chunks.0b818386e736f02b is valid.
cached chunk index 0b818386e736f02b gets merged...
trying to load cache/chunks.183453c5584142bc from the repo...
cache/chunks.183453c5584142bc is valid.
```
Only special stuff is my use of borg with backupname {hostname}.{user}._{utcnow} and set the environment variable  BORG_FILES_CACHE_SUFFIX=root

**Reply by ThomasWaldmann** [reply]
> borgstore.backends.errors.ObjectNotFound: cache/chunks[_hash]

These 2 exceptions are expected (b14 always tries to delete these outdated cache objects to clean up).

You only see them because you have debug logging activated and there is no special code suppressing them. Will have a look whether this can be improved...
